src/crypto/tls/tls_cf.go

```diff
@@ -48,8 +56,144 @@ func init() {
 
 // CFEvent is a value emitted at various points in the handshake that is
 // handled by the callback Config.CFEventHandler.
-//
-// NOTE: This API is EXPERIMENTAL and subject to change.
 type CFEvent interface {
 	Name() string
 }
+
+// CFEventTLS13ClientHandshakeTimingInfo carries intra-stack time durations
+// for TLS 1.3 state machine changes. It can be used for tracking metrics during a
+// connection. Some durations may be sensitive, such as the amount of time to
+// process a particular handshake message, so this event should only be used
+// for experimental purposes.
+type CFEventTLS13ClientHandshakeTimingInfo struct {
+	timer                   func() time.Time
+	start                   time.Time
+	WriteClientHello        time.Duration
+	ProcessServerHello      time.Duration
+	ReadEncryptedExtensions time.Duration
+	ReadCertificate         time.Duration
```

Comment:
Is `ReadCertificate` both used for client and server?

Reply:
There are two events, one that is emitted by the client and one that is emitted by the server. Both record this value.